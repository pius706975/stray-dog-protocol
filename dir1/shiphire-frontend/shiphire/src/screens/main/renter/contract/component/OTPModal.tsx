import React from 'react';
import { Modal, Pressable } from 'react-native';
import { Text, View } from 'react-native-ui-lib';
import { useDispatch } from 'react-redux';
import { Button, CustomText } from '../../../../../components';
import {
    CloseIcon,
    Color,
    FontFamily,
    FontSize,
    OTPIcon,
} from '../../../../../configs';
import {
    useSendOTPSignContract,
    useVerifySignContractOTP,
} from '../../../../../hooks';
import { modalSlice } from '../../../../../slices';
import { OTPModalContractProps } from '../../../../../types';
import { handleAxiosError } from '../../../../../utils';
import OTPField from './OTPField';
import { useTranslation } from 'react-i18next';

const OTPModal: React.FC<OTPModalContractProps> = ({
    visible,
    setVisible,
    parentProp,
    rentalId,
    renterEmail,
}) => {
    const { t } = useTranslation('common');
    const dispatch = useDispatch();
    const { hideModal, showModal } = modalSlice.actions;
    const mutationSendOTP = useSendOTPSignContract();
    const mutationVerifyOTP = useVerifySignContractOTP();
    const [pinReady, setPinReady] = React.useState<boolean>(false);
    const [countdown, setCountdown] = React.useState<number>(0);
    const [code, setCode] = React.useState<number>(0);
    const [countdownActive, setCountdownActive] =
        React.useState<boolean>(false);
    const navigation = parentProp.navigation;

    const startCountdown = () => {
        setCountdown(60);
        setCountdownActive(true);
        mutationSendOTP.mutate(undefined, {
            onSuccess: () => {},
            onError: err => {
                handleAxiosError(err);
            },
        });
    };

    const handleVerifyPressed = () => {
        let errorText = 'Request Failed';
        mutationVerifyOTP.mutate(
            { signContractOTP: code, rentalId },
            {
                onSuccess: () => {
                    setVisible(!visible);
                    navigation.replace('TransactionStatusTab');
                    dispatch(
                        showModal({
                            status: 'success',
                            text: t('OTPModal.successContractSign'),
                        }),
                    );
                    setTimeout(() => {
                        dispatch(hideModal());
                    }, 4000);
                },
                onError: err => {
                    handleAxiosError(err);
                    if (err.response?.status === 422) {
                        errorText = t('OTPModal.failedInvalidOTP');
                    }
                    dispatch(showModal({ status: 'failed', text: errorText }));
                    setTimeout(() => {
                        dispatch(hideModal());
                    }, 4000);
                    setCode(0);
                },
            },
        );
    };

    React.useEffect(() => {
        let countdownInterval: any;
        if (countdownActive) {
            countdownInterval = setInterval(() => {
                setCountdown(prevCountdown => prevCountdown - 1);
            }, 1000);
        }

        if (countdown === 0 || !countdownActive) {
            clearInterval(countdownInterval);
            setCountdownActive(false);
        }

        return () => {
            clearInterval(countdownInterval);
        };
    }, [countdown, countdownActive]);
    console.log('masuk');

    return (
        <Modal
            animationType="slide"
            transparent={true}
            visible={visible}
            testID="OTPModal">
            <View
                flex
                row
                center
                style={{
                    backgroundColor: 'rgba(0,0,0,0.7)',
                    paddingHorizontal: 10,
                }}>
                <Pressable
                    style={{
                        borderRadius: 10,
                        paddingVertical: 30,
                        backgroundColor: Color.bgColor,
                        flex: 1,
                        alignItems: 'center',
                    }}>
                    <Pressable
                        testID="pressVisible"
                        style={{
                            position: 'absolute',
                            right: '6%',
                            top: '3%',
                        }}
                        onPress={() => setVisible(!visible)}>
                        <CloseIcon />
                    </Pressable>
                    <OTPIcon />
                    <CustomText
                        fontSize="xl"
                        fontFamily="bold"
                        color="darkTextColor"
                        lineHeight={50}>
                        {t('OTPModal.textOTPVerif')}
                    </CustomText>
                    <CustomText
                        fontSize="sm"
                        fontFamily="regular"
                        color="darkTextColor">
                        {t('OTPModal.textEnterOTP')}
                    </CustomText>
                    <CustomText
                        fontSize="sm"
                        fontFamily="bold"
                        color="darkTextColor">
                        {renterEmail}
                    </CustomText>
                    <OTPField
                        testId="OTPField"
                        setPinReady={setPinReady}
                        setModalCode={setCode}
                        modalCode={code}
                    />
                    <Text
                        style={{
                            fontSize: FontSize.sm,
                            fontFamily: FontFamily.semiBold,
                            color: Color.darkTextColor,
                        }}>
                        {t('OTPModal.textDidnt')}
                    </Text>
                    {countdown === 0 ? (
                        <Pressable onPress={startCountdown} testID="pressCount">
                            <Text
                                style={{
                                    fontSize: FontSize.sm,
                                    fontFamily: FontFamily.semiBold,
                                    color: Color.primaryColor,
                                    borderBottomWidth: 1,
                                    borderColor: Color.primaryColor,
                                }}>
                                {t('OTPModal.textResend')}
                            </Text>
                        </Pressable>
                    ) : (
                        <Text
                            style={{
                                fontFamily: FontFamily.regular,
                                fontSize: FontSize.sm,
                                color: Color.darkTextColor,
                            }}>
                            {t('OTPModal.textResendAgain')}{' '}
                            <Text
                                style={{
                                    fontFamily: FontFamily.semiBold,
                                    color: Color.primaryColor,
                                }}>
                                {countdown}s
                            </Text>
                        </Text>
                    )}
                    <View row marginT-20 paddingH-30>
                        <View flex>
                            <Button
                                testID="verifyButton"
                                title={t('OTPModal.labelButtonVerify')}
                                disable={!pinReady}
                                onSubmit={handleVerifyPressed}
                                isSubmitting={mutationVerifyOTP.isLoading}
                            />
                        </View>
                    </View>
                </Pressable>
            </View>
        </Modal>
    );
};

export default OTPModal;
