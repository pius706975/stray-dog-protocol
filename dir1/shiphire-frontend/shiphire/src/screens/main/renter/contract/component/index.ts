export { default as OTPField } from "./OTPField"
export { default as OTPModal } from "./OTPModal"