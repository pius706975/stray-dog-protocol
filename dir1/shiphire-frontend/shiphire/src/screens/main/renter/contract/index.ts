export { default as ContractPreview } from "./ContractPreview"