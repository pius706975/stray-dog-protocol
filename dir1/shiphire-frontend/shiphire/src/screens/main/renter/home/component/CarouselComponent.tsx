import React from 'react';
import { Pressable } from 'react-native';
import { Image, View } from 'react-native-ui-lib';
import {
    CategoryContainer,
    CustomText,
    StarRating,
} from '../../../../../components';
import { Color } from '../../../../../configs';
import { CarouselComponentProps } from '../../../../../types';

const CarouselComponent: React.FC<CarouselComponentProps> = ({
    categoryLabel,
    imageUrl,
    onPress,
    shipName,
    shipRating,
    testId,
}) => {
    return (
        <Pressable style={{ flex: 1 }} onPress={onPress} testID={testId}>
            <Image
                overlayType={Image.overlayTypes.BOTTOM}
                style={{ flex: 3, borderRadius: 10 }}
                source={{
                    uri: imageUrl
                        ? imageUrl
                        : 'https://picsum.photos/id/237/200/300',
                }}
            />
            <View
                flex
                style={{
                    position: 'absolute',
                    width: '100%',
                    height: '100%',
                    marginTop: 8,
                    marginRight: 8,
                }}>
                <View row spread>
                    <View />
                    <CategoryContainer
                        testId={`categoryPressable-${shipName}`}
                        label={categoryLabel}
                        onPress={() => console.log('pressed')}
                    />
                </View>
            </View>
            <View
                flex
                centerV
                row
                spread
                paddingH-10
                paddingB-20
                style={{
                    backgroundColor: Color.primaryColor,
                    borderBottomRightRadius: 10,
                    borderBottomLeftRadius: 10,
                }}>
                <CustomText
                    fontFamily="semiBold"
                    fontSize="xl"
                    color="secColor">
                    {shipName}
                </CustomText>
                <StarRating rating={shipRating} />
            </View>
        </Pressable>
    );
};

export default CarouselComponent;
