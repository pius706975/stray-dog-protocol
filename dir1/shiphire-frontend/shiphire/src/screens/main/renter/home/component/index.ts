export { default as CarouselComponent } from "./CarouselComponent"
export { default as HeaderHome } from "./HeaderHome"
export { default as ShipCategory } from "./ShipCategory"
export { default as ShipListView } from "./ShipListView"
export { default as TopRateShipComponent } from "./TopRateShipComponent"