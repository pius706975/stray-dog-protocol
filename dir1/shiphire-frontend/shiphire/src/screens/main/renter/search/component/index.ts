export { default as DatePickerFilter } from './DatePickerFilter';
