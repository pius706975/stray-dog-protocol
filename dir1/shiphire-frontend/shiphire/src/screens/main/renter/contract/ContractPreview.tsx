import React from 'react';
import { View } from 'react-native-ui-lib';
import PDFView from 'react-native-view-pdf';
import { useDispatch } from 'react-redux';
import { Button, ScreenLayout } from '../../../../components';
import {
    CheckIcon,
    Color,
    EditIcon,
    USERDATA,
    getDataFromLocalStorage,
} from '../../../../configs';
import { useSendOTPSignContract } from '../../../../hooks';
import { progressIndicatorSlice } from '../../../../slices';
import { ContractPreviewProps } from '../../../../types';
import { handleAxiosError } from '../../../../utils';
import { OTPModal } from './component';
import { useTranslation } from 'react-i18next';

const ContractPreview: React.FC<ContractPreviewProps> = props => {
    const dispatch = useDispatch();
    const { contractUrl, rentalId } = props.route.params;
    const { t } = useTranslation('common');
    const mutationSendOTP = useSendOTPSignContract();
    const { showProgressIndicator, hideProgressIndicator } =
        progressIndicatorSlice.actions;
    const [visible, setVisible] = React.useState<boolean>(false);
    const [renterEmail, setRenterEmail] = React.useState<string>('');
    const [contractLoading, setContractLoading] =
        React.useState<boolean>(false);

    const handleAcceptPressed = () => {
        mutationSendOTP.mutate(undefined, {
            onSuccess: () => {
                setVisible(!visible);
            },
            onError: err => {
                handleAxiosError(err);
            },
        });
    };

    React.useEffect(() => {
        getDataFromLocalStorage(USERDATA).then(resp => {
            setRenterEmail(resp.email);
        });
        setContractLoading(true);
    }, []);

    React.useEffect(() => {
        dispatch(showProgressIndicator());
        setTimeout(() => {
            dispatch(hideProgressIndicator());
        }, 1000);
    }, [contractLoading]);

    return (
        <ScreenLayout
            testId="ContractPreviewScreen"
            backgroundColor="light"
            padding={10}>
            {contractUrl && (
                <PDFView
                    fadeInDuration={250.0}
                    style={{
                        flex: 1,
                        borderWidth: 1,
                        backgroundColor: Color.bgColor,
                    }}
                    resource={contractUrl}
                    resourceType={'url'}
                    onLoad={() => {
                        setContractLoading(false);
                        setTimeout(() => {
                            dispatch(hideProgressIndicator());
                        }, 1000);
                    }}
                    onError={error => console.log('Cannot render PDF', error)}
                />
            )}
            <View
                row
                spread
                center
                flex
                style={{
                    position: 'absolute',
                    bottom: 30,
                }}>
                {/* <View flex row center>
                    <Button
                        title={t('ContractPreview.labelButtonNegotiate')}
                        onSubmit={() => {}}
                        color="warning"
                        leftIcon={<EditIcon />}
                    />
                </View> */}
                <View flex row right>
                    <Button
                        testID="acceptButton"
                        title={t('ContractPreview.labelButtonAcceptContract')}
                        onSubmit={handleAcceptPressed}
                        color="success"
                        leftIcon={<CheckIcon />}
                        isSubmitting={mutationSendOTP.isLoading}
                    />
                </View>
            </View>
            {visible && (
                <OTPModal
                    visible={visible}
                    setVisible={setVisible}
                    parentProp={props}
                    rentalId={rentalId}
                    renterEmail={renterEmail}
                />
            )}
        </ScreenLayout>
    );
};

export default ContractPreview;
