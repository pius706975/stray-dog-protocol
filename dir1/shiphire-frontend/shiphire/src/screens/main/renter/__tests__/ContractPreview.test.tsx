import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import ContractPreview from '../contract/ContractPreview';
import { TOKEN, USERDATA } from '../../../../configs';
import AsyncStorage from '@react-native-async-storage/async-storage';
import { QueryClient, QueryClientProvider } from 'react-query';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { MainScreenParamList, MainStackParamList } from '../../../../types';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import httpRequest from '../../../../services/api';
import MockAdapter from 'axios-mock-adapter';
import { Provider } from 'react-redux';
import store from '../../../../store';
import { NavigationContainer } from '@react-navigation/native';
import PDFView from 'react-native-view-pdf';
import { act } from 'react-test-renderer';
import { OTPModal } from '../contract/component';

const queryClient = new QueryClient();
const MainStack = createNativeStackNavigator<MainStackParamList>();
const mockAdapter = new MockAdapter(httpRequest);

jest.mock('react-i18next', () => ({
    useTranslation: () => ({
        t: key => key,
    }),
}));

const mockParams = {
    contractUrl: 'https://example.com/contract.pdf',
    rentalId: '123',
};

const mockContractComponent = () => (
    <QueryClientProvider client={queryClient}>
        <Provider store={store}>
            <NavigationContainer>
                <MainStack.Navigator>
                    <MainStack.Screen
                        name="ContractPreview"
                        component={ContractPreview}
                        initialParams={mockParams}
                    />
                </MainStack.Navigator>
            </NavigationContainer>
        </Provider>
    </QueryClientProvider>
);

AsyncStorage.setItem(
    USERDATA,
    JSON.stringify({
        name: 'Usmanul',
        email: 'usman@email.com',
        password: 'test',
        phoneNumber: '12313123',
        isVerified: true,
        isCompanySubmitted: true,
    }),
);
AsyncStorage.setItem(
    TOKEN,
    JSON.stringify({
        accessToken: 'test',
        refreshToken: 'test',
    }),
);

describe('contract Preview', () => {
    it('should render the screen correctly', async () => {
        const contractPreview = render(mockContractComponent()).toJSON();
        expect(contractPreview).toMatchSnapshot();
    });
});
describe('Receipt Preview', () => {
    it('should render the contract', async () => {
        render(mockContractComponent());
        const pdfView = render(
            <PDFView
                resource="https://example.com/contract.pdf"
                resourceType="url"
            />,
        );
        await act(() => {
            expect(pdfView).toBeTruthy();
        });
    });
    it('should not render the PDF view if contract is not provided', async () => {
        render(mockContractComponent());
        const pdfView = render(
            <PDFView
                resource={''}
                onError={error => {
                    console.log('Cannot render PDF', error);
                }}
            />,
        );

        const progressState = store.getState().progressIndicator;
        expect(progressState).toEqual({
            visible: true,
        });
        await waitFor(() => {
            const progressState = store.getState().progressIndicator;
            expect(progressState).toEqual({
                visible: false,
            });
        });
        await act(async () => {
            expect(pdfView).toBeDefined();
        });
    });
    it('should call handleAcceptPressed when the accept button is pressed', async () => {
        const { getByTestId } = render(mockContractComponent());
        const acceptButton = getByTestId('acceptButton');
        fireEvent.press(acceptButton);

        await act(() => {
            expect(acceptButton).toBeDefined();
        });

        mockAdapter
            .onPost('/renter/send-otp-sign-contract')
            .reply(200, { status: 'success' });

        await act(async () => {
            expect(acceptButton).toBeDefined();
        });
    });
    it('should call OTPModal when the accept button is pressed', async () => {
        const { getByTestId } = render(mockContractComponent());
        const acceptButton = getByTestId('acceptButton');
        fireEvent.press(acceptButton);

        await act(() => {
            expect(acceptButton).toBeDefined();
        });

        mockAdapter
            .onPost('/renter/send-otp-sign-contractl')
            .reply(200, { status: 'success' });

        await act(async () => {
            expect(acceptButton).toBeDefined();
        });

        const modal = getByTestId('OTPModal');

        expect(modal).toBeTruthy();
    });
    it('should call OTPModal when the accept button is pressed and render the component', async () => {
        const { getByTestId, getByText } = render(mockContractComponent());
        const acceptButton = getByTestId('acceptButton');
        fireEvent.press(acceptButton);

        await act(() => {
            expect(acceptButton).toBeDefined();
        });

        mockAdapter
            .onPost('/renter/send-otp-sign-contract')
            .reply(200, { status: 'success' });

        await act(async () => {
            expect(acceptButton).toBeDefined();
        });

        const modal = getByTestId('OTPModal');

        expect(modal).toBeTruthy();

        const pressVisible = getByTestId('pressVisible');

        expect(pressVisible).toBeDefined();

        expect(getByText('OTPModal.textOTPVerif')).toBeTruthy();

        const press = getByTestId('pressCount');

        expect(press).toBeDefined();
    });
    it('should send OTP again after click send code', async () => {
        const { getByTestId, getByText } = render(mockContractComponent());
        const acceptButton = getByTestId('acceptButton');
        fireEvent.press(acceptButton);

        await act(() => {
            expect(acceptButton).toBeDefined();
        });

        const modal = getByTestId('OTPModal');

        expect(modal).toBeTruthy();

        const pressResend = getByTestId('pressCount');
        fireEvent.press(pressResend);

        // Assert that another OTP request has been sent
        await waitFor(() => {
            mockAdapter
                .onPost('/renter/send-otp-sign-contract')
                .reply(200, { status: 'success' });
        });
    });
    it('should call OTPModal when the accept button is pressed and input the field', async () => {
        jest.useFakeTimers();

        const { getByTestId } = render(mockContractComponent());
        const acceptButton = getByTestId('acceptButton');
        fireEvent.press(acceptButton);

        await act(() => {
            expect(acceptButton).toBeDefined();
        });

        mockAdapter
            .onPost('/renter/send-otp-sign-contract')
            .reply(200, { status: 'success' });

        await act(async () => {
            expect(acceptButton).toBeDefined();
        });

        const modal = getByTestId('OTPModal');

        expect(modal).toBeTruthy();

        const input = getByTestId('OTPInput');

        fireEvent.changeText(input, '1234');

        expect(input.props.value).toEqual('1234');

        const verifyButton = getByTestId('verifyButton');

        fireEvent.press(verifyButton);

        await act(async () => {
            expect(verifyButton).toBeDefined();
        });

        mockAdapter
            .onPost('/renter/verify-otp-sign-contract')
            .reply(200, { status: 'success' });

        await act(async () => {
            expect(verifyButton).toBeDefined();
        });
    });
    it('should success sign the contract', async () => {
        jest.useFakeTimers();

        const { getByTestId } = render(mockContractComponent());
        const acceptButton = getByTestId('acceptButton');
        fireEvent.press(acceptButton);

        await act(() => {
            expect(acceptButton).toBeDefined();
        });

        mockAdapter
            .onPost('/renter/send-otp-sign-proposal')
            .reply(200, { status: 'success' });

        await act(async () => {
            expect(acceptButton).toBeDefined();
        });

        const modal = getByTestId('OTPModal');

        expect(modal).toBeTruthy();

        const input = getByTestId('OTPInput');

        fireEvent.changeText(input, '1234');

        expect(input.props.value).toEqual('1234');

        const verifyButton = getByTestId('verifyButton');

        fireEvent.press(verifyButton);

        await act(async () => {
            expect(verifyButton).toBeDefined();
        });

        mockAdapter
            .onPost('/renter/verify-otp-sign-contract')
            .reply(200, { status: 'success' });

        expect(verifyButton).toBeDefined();

        await waitFor(() => {
            const modalState = store.getState().modal;

            jest.advanceTimersByTime(4000);
            expect(modalState).toEqual({
                visible: false,
                status: undefined,
                text: '',
            });
        });
    });
    it('should verify the OTP', async () => {
        const { getByTestId } = render(mockContractComponent());
        const acceptButton = getByTestId('acceptButton');
        await act(async () => {
            fireEvent.press(acceptButton);
        });

        await waitFor(() => {
            expect(getByTestId('OTPModal')).toBeDefined();
        });

        mockAdapter
            .onPost('/renter/send-otp-sign-contract')
            .reply(200, { status: 'success' });

        await act(async () => {
            expect(acceptButton).toBeDefined();
        });

        mockAdapter
            .onPost('/renter/verify-otp-sign-contract')
            .reply(200, { status: 'success' });
    });
});

beforeEach(() => {
    jest.clearAllMocks();
});

describe('error handling', () => {
    it('should error verify OTP', async () => {
        jest.useFakeTimers();

        const { getByTestId } = render(mockContractComponent());
        const acceptButton = getByTestId('acceptButton');
        fireEvent.press(acceptButton);

        await act(() => {
            expect(acceptButton).toBeDefined();
        });

        mockAdapter
            .onPost('/renter/send-otp-sign-contract')
            .reply(200, { status: 'success' });

        await act(async () => {
            expect(acceptButton).toBeDefined();
        });

        const modal = getByTestId('OTPModal');

        expect(modal).toBeTruthy();

        const input = getByTestId('OTPInput');

        fireEvent.changeText(input, '124');

        const verifyButton = getByTestId('verifyButton');

        fireEvent.press(verifyButton);

        await act(async () => {
            expect(verifyButton).toBeDefined();
        });

        mockAdapter
            .onPost('/renter/verify-otp-sign-contract')
            .reply(422, { status: 'failed' });

        expect(verifyButton).toBeDefined();

        // await waitFor(() => {
        //     const modalState = store.getState().modal;
        //     jest.advanceTimersByTime(4000);
        //     expect(modalState).toEqual({
        //         visible: true,
        //         status: 'failed',
        //         text: 'OTPModal.failedInvalidOTP',
        //     });
        // });
        // const updatedModalState = store.getState().modal;
        // expect(updatedModalState.visible).toBe(true);
    });
});
