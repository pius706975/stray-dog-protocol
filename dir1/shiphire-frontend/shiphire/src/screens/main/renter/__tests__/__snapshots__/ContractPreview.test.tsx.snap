// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contract Preview should render the screen correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                undefined,
                undefined,
                {
                  "flex": 1,
                },
                undefined,
                {},
                {},
                {},
                {
                  "backgroundColor": "#f0f6fb",
                  "justifyContent": "space-between",
                  "position": "relative",
                },
              ]
            }
            testID="ContractPreviewScreen"
          >
            <RCTScrollView
              contentContainerStyle={
                {
                  "alignItems": undefined,
                  "flex": undefined,
                  "flexGrow": 1,
                  "gap": undefined,
                  "justifyContent": undefined,
                  "marginBottom": undefined,
                  "padding": 10,
                  "paddingVertical": undefined,
                }
              }
            >
              <View>
                <View
                  center={true}
                  row={true}
                  spread={true}
                  style={
                    [
                      undefined,
                      undefined,
                      {
                        "flex": 1,
                      },
                      undefined,
                      {},
                      {},
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      },
                      {
                        "bottom": 30,
                        "position": "absolute",
                      },
                    ]
                  }
                >
                  <View
                    row={true}
                    style={
                      [
                        undefined,
                        undefined,
                        {
                          "flex": 1,
                        },
                        undefined,
                        {},
                        {},
                        {
                          "flexDirection": "row",
                          "justifyContent": "flex-end",
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": undefined,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "alignItems": "center",
                          "backgroundColor": "#1fcc44",
                          "borderRadius": 6,
                          "elevation": 2,
                          "flexDirection": "row",
                          "gap": 10,
                          "justifyContent": "center",
                          "paddingHorizontal": 10,
                          "paddingVertical": 6,
                        }
                      }
                      testID="acceptButton"
                    >
                      <RNSVGSvgView
                        align="xMidYMid"
                        bbHeight="24"
                        bbWidth="24"
                        fill="none"
                        focusable={false}
                        height={24}
                        meetOrSlice={0}
                        minX={0}
                        minY={0}
                        style={
                          [
                            {
                              "backgroundColor": "transparent",
                              "borderWidth": 0,
                            },
                            {
                              "flex": 0,
                              "height": 24,
                              "width": 24,
                            },
                          ]
                        }
                        vbHeight={24}
                        vbWidth={24}
                        width={24}
                        xml="<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
        <g clip-path="url(#clip0_77_30622)">
        <path d="M9.00001 16.1703L5.53001 12.7003C5.14002 12.3103 4.51001 12.3103 4.12001 12.7003C3.73001 13.0903 3.73001 13.7203 4.12001 14.1103L8.30001 18.2903C8.69001 18.6803 9.32001 18.6803 9.71001 18.2903L20.29 7.71031C20.68 7.32031 20.68 6.69031 20.29 6.30031C19.9 5.91031 19.27 5.91031 18.88 6.30031L9.00001 16.1703Z" fill="#f0f6fb"/>
        </g>
        <defs>
        <clipPath id="clip0_77_30622">
        <rect width="24" height="24" fill="white"/>
        </clipPath>
        </defs>
        </svg>
        "
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <RNSVGGroup
                          fill={null}
                          propList={
                            [
                              "fill",
                            ]
                          }
                        >
                          <RNSVGGroup
                            clipPath="clip0_77_30622"
                            fill={
                              {
                                "payload": 4278190080,
                                "type": 0,
                              }
                            }
                          >
                            <RNSVGPath
                              d="M9.00001 16.1703L5.53001 12.7003C5.14002 12.3103 4.51001 12.3103 4.12001 12.7003C3.73001 13.0903 3.73001 13.7203 4.12001 14.1103L8.30001 18.2903C8.69001 18.6803 9.32001 18.6803 9.71001 18.2903L20.29 7.71031C20.68 7.32031 20.68 6.69031 20.29 6.30031C19.9 5.91031 19.27 5.91031 18.88 6.30031L9.00001 16.1703Z"
                              fill={
                                {
                                  "payload": 4293981947,
                                  "type": 0,
                                }
                              }
                              propList={
                                [
                                  "fill",
                                ]
                              }
                            />
                          </RNSVGGroup>
                          <RNSVGDefs>
                            <RNSVGClipPath
                              fill={
                                {
                                  "payload": 4278190080,
                                  "type": 0,
                                }
                              }
                              name="clip0_77_30622"
                            >
                              <RNSVGRect
                                fill={
                                  {
                                    "payload": 4294967295,
                                    "type": 0,
                                  }
                                }
                                height="24"
                                propList={
                                  [
                                    "fill",
                                  ]
                                }
                                width="24"
                                x="0"
                                y="0"
                              />
                            </RNSVGClipPath>
                          </RNSVGDefs>
                        </RNSVGGroup>
                      </RNSVGSvgView>
                      <Text
                        numberOfLines={0}
                        style={
                          {
                            "borderBottomWidth": 0,
                            "borderColor": undefined,
                            "color": "#f0f6fb",
                            "fontFamily": "Kanit-Medium",
                            "fontSize": 18,
                            "lineHeight": undefined,
                            "textAlign": undefined,
                          }
                        }
                      >
                        ContractPreview.labelButtonAcceptContract
                      </Text>
                      <View
                        style={
                          [
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            {
                              "position": "absolute",
                              "right": 20,
                            },
                          ]
                        }
                      />
                    </View>
                  </View>
                </View>
              </View>
            </RCTScrollView>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="ContractPreview"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
